web/pdf_thumbnail_view.js

```diff
@@ -452,8 +438,20 @@ class PDFThumbnailView {
     this._convertCanvasToImage();
   }
 
-  get pageId() {
-    return this.pageLabel !== null ? this.pageLabel : this.id;
+  get _thumbPageTitle() {
+    return this.l10n.get(
```

Comment:
Can we use the shadow utility so we only fetch this once? We have multiple call sites and the label should always be the same for the same thumbnail view, so this might make it a bit faster. Same below.

Reply:
Nope, since these will generally be called more than once for documents with page labels :-)

Reply:
Good point!